### DISCUSSION TITLE
Cache Guid over multiple Verifys

### DISCUSSION THREAD

**Discussion by LodewijkSioen** [VerifyTests/Verify#919] [Q&A]
I'm writing integration tests with multiple steps. I would like to have the Guids to be re-used from one step to the other so I'm sure that all Verify steps use the same replacement for the same Guid.

eg:
```
await VerifyJson("{id: 'abc123', reference: 'abc456'}"); //-> {id: 'Guid_1', reference: 'Guid_2'}
await VerifyJson("{id: 'def123', reference: 'abc456'}"); //-> {id: 'Guid_3', reference: 'Guid_2'}
```

Is such a thing possible?

**Comment by LodewijkSioen** [comment]
After diving in the code: What I'm looking for is a way to broaden the scope of the `Counter` so it can be used for multiple `Verify`s.

**Comment by SimonCropp** [comment]
Verify is not designed to have multiple verifies happen in the same method. 

the code you showed should throw an exception

> The prefix has already been used

**Reply by LodewijkSioen** [reply]
Yes, it was a minimal sample. In reality it's more like
```
//work is done here, which returns the json
await VerifyJson("{id: 'abc123', reference: 'abc456'}", settings)
            .UseFileName("Step1");
// More work happens here, which returns another json
await VerifyJson("{id: 'def123', reference: 'abc456'}", settings)
            .UseFileName("Step2");

```
If Verify not designed to work like this, what problems can I run into by (ab)using it like this?

**Reply by SimonCropp** [reply]
> what problems can I run into by (ab)using it like this?

well for example the behavior you are describing in this discussion

**Reply by LodewijkSioen** [reply]
owkay, I'll write my own scrubber that keeps count and see what else blows up in my face.

Thanks for this library, btw. I can now throw away piles of code 👍

**Comment by SimonCropp** [comment]
how about this as an alternative

        await Verify(
            new
            {
                step1 = JToken.Parse("{'one': {'value': '654f48ed-7a41-4a62-b317-e22f7b7c139b'}}"),
                step2 = JToken.Parse("{'two': {'value': '654f48ed-7a41-4a62-b317-e22f7b7c139b'}}")
            });

**Reply by LodewijkSioen** [reply]
So do all the work first, gather up the results and verify them in one big batch. That could work, but I'de rather have my test fail if step 1 produces unwanted results. The diff would be unusable after that anyway.

Thanks for the suggestion.
